src/hooks/useClickOutside.ts

```diff
@@ -4,22 +4,34 @@ import { useEffect, useRef } from 'react'
  * Custom hook that fires an onClick callback when the user clicks
  * outside of the element referenced by the returned ref.
  *
- * @param {function} onClick the callback fired when a click is performed "outside"
+ * @param onClick the callback fired when a click is performed "outside"
+ * @param isException optional check to be run on the element that receives
+ * the "click." If isException returns true, the click is not considered outside
+ * even if it was outside the element referenced.
+  
  * @returns a ref that should be used to mark the "inside" element
  */
-const useClickOutside = (onClick: () => void) => {
+const useClickOutside = (
+  onClick: () => void,
+  isException: (target: EventTarget) => boolean = () => false
```

Comment:
naming bikeshed: maybe this could be something like `ignoreClick`, just so we avoid the 'exception' namespace collision w JS exceptions?